packages/manager/modules/pci/src/projects/new/components/voucher/voucher.class.js

```diff
@@ -0,0 +1,47 @@
+import get from 'lodash/get';
+
+import { VOUCHER_CODE_ENUM } from './constants';
+
+export default class PciVoucher {
+  constructor(options = {}) {
+    // options available from APIs
+    this.value = options.value || null;
+    this.valid = options.valid || false;
+    this.paymentMethodRequired = options.paymentMethodRequired || false;
+    this.credit = options.credit || null;
+
+    // other options
+    this.error = options.error || null;
+  }
+
+  setInfos(infos) {
+    // reset before
+    this.reset();
+
+    if (infos.error) {
+      // set validity
+      this.valid = false;
+      // set error info
+      this.error = {
+        status: infos.error,
+        statusText: get(VOUCHER_CODE_ENUM, infos.error, 'DEFAULT'),
+      };
+    } else {
+      // set validity
+      this.valid = true;
+      // set other infos
+      Object.assign(this, infos);
+    }
+
+    return this;
+  }
+
+  reset() {
+    this.valid = false;
+    this.paymentMethodRequired = false;
+    this.credit = null;
+    this.error = null;
+
+    return this;
+  }
+}
```

Comment:
Imo, you should be able to create a voucher like 
```suggestion
 const voucher = new Voucher(...)
```
and call a isValid function
```suggestion
 voucher.isValid() 
```
and the logic is handled by the voucher class. 
Also, I found it to reset the info that you're supposed to give in the constructor